src/reactviews/pages/ExecutionPlan/properties.tsx

```diff
@@ -0,0 +1,638 @@
+    const handleSort = async (sortOption: ep.SortOption) => {
+        const currentItems = resetFiltering();
+        setItems(
+            recursiveSort(
+                currentItems,
+                currentItems.filter((item) => !item.isChild),
+                sortOption,
+            ),
+        );
+    };
+
+    const handleExpandAll = async () => {
+        const currentItems = resetFiltering();
+        setOpenedButtons(currentItems.map((item) => item.name));
+        setShownChildren(currentItems.map((item) => item.id));
+    };
+
+    const handleCollapseAll = async () => {
+        resetFiltering();
+        setOpenedButtons([]);
+        setShownChildren([]);
+    };
+
+    const handleFilter = async (searchValue: string) => {
+        // on starting filtering, save the current items so that when filtering stops
+        // it can properly reset the items
+        let firstFilter = false;
+        if (items.length === numItems) {
+            setUnfilteredItems(items);
+            firstFilter = true;
+        }
+
+        if (searchValue !== "") {
+            // react updates state asynchronously, so if the state of unfiltered
+            // items hasn't been updated yet, ie. on the first filter, use items instead
+            const currentItems = firstFilter ? items : unfilteredItems;
+            let filteredItems = currentItems.filter(
+                (item) =>
+                    item.name.includes(searchValue) ||
+                    item.value.includes(searchValue),
+            );
+
+            setItems(
+                buildFilteredItemsFromChildList(filteredItems, currentItems),
+            );
+            setIsFiltered(true);
+        }
+        // filtering is removed
+        else {
+            resetFiltering();
+        }
+    };
+
+    function resetFiltering(): ep.ExecutionPlanPropertyTableItem[] {
+        // react updates state asynchronously, so if the state of unfiltered
+        // items hasn't been updated yet, ie. sorting while actively filtering,
+        // use unfiltered items instead
+        const currentItems = isFiltered ? unfilteredItems : items;
+        setItems(unfilteredItems);
+        setIsFiltered(false);
+        setInputValue("");
+        return currentItems;
+    }
+
+    const columns: TableColumnDefinition<ep.ExecutionPlanPropertyTableItem>[] =
+        [
+            createTableColumn<ep.ExecutionPlanPropertyTableItem>({
+                columnId: "name",
+                renderHeaderCell: () => NAME,
+                renderCell: (item) => (
+                    // Add tabbing based on the "level" of the item in the table,
+                    // and add expand button based on whether the item has children
+                    <TableCellLayout truncate className={classes.textContainer}>
+                        {`\u200b\t`.repeat(item.level * 6)}
+                        {item.children.length > 0 && (
+                            <Button

```

Comment:
There are man  changes we need to make in order to make tree-grid accessible. That includes adding special keyboard shortcuts and aria attributes. 

Please create an issue that tracks it and follow it up on a separate PR. 

Maybe we should think about creating a generic component for TreeGrid so that every thing that use tree grid can get the same level accessibility without reimposed

Reply:
made this issue for it: #18051